3.5/programs-arangobackup-limitations.md

```diff
@@ -0,0 +1,155 @@
+This means that one cannot restore a 3-node ArangoDB cluster's hot backup to
+any other deployment than another 3-node ArangoDB cluster of the same version.
+
+RocksDB Storage Engine Only
+---------------------------
+
+Hot backups rely on creation of hard links on actual RocksDB data files and
+directories. The same or according file system level mechanisms are not
+available to MMFiles deployments.
+
+Storage Space
+-------------
+
+Without the creation of hot backups, RocksDB keeps compacting the file system
+level files as the operation continues. Compacted files are subsequently
+deleted automatically. Every hot backup needs to hold on to the
+files as they were at the moment of the hot backup creation, thus preventing
+the deletions and consequently growing the storage space of the ArangoDB
+data directory. That growth of course depends on the amount of write operations
+per time.
+
+This is a crucial factor for sustained operation and might require
+significantly higher storage reservation for ArangoDB instances involved and
+a much more fine grained monitoring of storage usage than before.
+
+Also note that in a cluster each RocksDB instance will be backed up
+individually and hence the overall storage space will be the sum of all
+RocksDB instances (i.e., data which is replicated between instances will
+not be de-duplicated for performance reasons).
+
+Global Transaction Lock
+-----------------------
+
+In order to be able to create consistent hot backups, it is mandatory to get
+a very brief global transaction lock across the entire installation.
+In single server deployments constant invocation of very long running
+transactions could prevent that from ever happening during a timeout period.
+The same holds true for clusters, where this lock must now be obtained on all
+database servers at the same time.
+
+Especially in the cluster the result of these successively longer tries to
+obtain the global transaction lock might become visible in periods of apparent
+dead time. Locks might be obtained on some machines and and not on others, so
+that the process has to be retried over and over. Every unsuccessful try would
+then lead to the release of all partial locks.
+
+ArangoSearch not Supported Yet
+------------------------------
+
+ArangoSearch views are not backed up and thus not restored yet.
+Therefore, views have to be dropped and recreated after a restore.
+This happens automatically in the background, but in particular in the
+presence of large amounts of data, the recreation of the ArangoSearch
+indexes can take some time after the restore. It is planned to rectify
+this limitation in one of the next releases.
+
+Note furthermore that a running query with views can prevent a restore
+operation from happening whilst the query is running.
+
+Windows not Supported Yet
+-------------------------
+
+The first release of the hot backup feature (3.5.1) does not support
+the Windows version of ArangoDB. 
```

Comment:
Is it planned for 3.5.2? If no, then we should avoid such a statement and just say that it's not supported yet.

Reply:
Changed to not mention specific versions